Hestia/app/src/main/java/hestia/UI/elements/DeviceBar.java

```diff
@@ -0,0 +1,210 @@
+package hestia.UI.elements;
+
+import android.app.Activity;
+import android.os.AsyncTask;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.widget.PopupMenu;
+import android.util.Log;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.CompoundButton;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+import android.widget.Switch;
+import android.widget.TextView;
+import android.widget.Toast;
+import com.rugged.application.hestia.R;
+import java.io.IOException;
+import hestia.UI.dialogs.ChangeNameDialog;
+import hestia.UI.dialogs.SlidersDialog;
+import hestia.backend.ServerCollectionsInteractor;
+import hestia.backend.exceptions.ComFaultException;
+import hestia.backend.models.Activator;
+import hestia.backend.models.ActivatorState;
+import hestia.backend.models.Device;
+
+/**
+ *  This class takes care of the deviceBar.
+ * The devicebar is the 'row' in the expandable list of a single device.
+ */
+
+public class DeviceBar extends RelativeLayout {
+    private Activity context;
+    private Device device;
+    private ServerCollectionsInteractor serverCollectionsInteractor;
+    private final static String TAG = "DeviceBar";
+    private FragmentManager fm;
+
+    public DeviceBar(FragmentManager fm, Activity context, Device device, ServerCollectionsInteractor serverCollectionsInteractor) {
+        super(context);
+        this.device = device;
+        this.serverCollectionsInteractor = serverCollectionsInteractor;
+        this.context = context;
+        this.fm = fm;
+        initView();
+    }
+
+    public void initView() {
```

Comment:
Try to reduce the complexity of these kind of methods, they're not particularly fun to read trough,